local-proxy/src/index.ts

```diff
@@ -1,22 +1,25 @@
 // tslint:disable:no-console
 import path from 'path';
-import net from 'net';
 import nodemon from 'nodemon';
-import proxy from 'http-proxy-middleware';
+import proxy from 'http-proxy';
 import { Application } from 'express';
 import nanoid from 'nanoid';
+import { EventEmitter } from 'events';
 
-export async function startProxy(rootDir: string, localToken: string) {
-  const server = net.createServer();
+const isTestEnv = process.env.NODE_ENV === 'test';
 
-  await new Promise((resolve, reject) => {
-    server.listen(0, '127.0.0.1', resolve);
-    server.once('error',  reject);
-    return server;
-  });
-  const { port } = server.address() as net.AddressInfo;
+function log(message?: any, ...optionalParams: any[]) {
+  if (!isTestEnv) {
```

Comment:
Don't understand: log _except_ during tests?

Reply:
Yep, tests should only output success or failure (in verbose mode also the list of tests). The justification is in https://github.com/binaris/shiftjs/blob/ee6e17d3f3ec371ebd9aa0ad2182160cc103bdc2/common/config/rush/command-line.json#L83.

Reply:
I think I can confirm what you're saying below.

If I'm reading correctly (ironically, this entire text about inadvertent output writing) gets highlighted like an error in GitHub...) this applies only to standard output.  Tests are still allowed to write to standard output, but then Ava moves standard output to standard error.  I think.